cairo/tests/utils/serde.py

```diff
@@ -427,28 +426,16 @@ def serialize_type(self, path: Tuple[str, ...], ptr) -> Any:
             return python_cls(value.to_bytes(python_cls.LENGTH, "little"))
 
         if python_cls == BNF12:
-            # BNF12 is represented as a struct with 12 UInt384
-            # We need to extract these values and create a BNF12 object
-            if value is None:
-                return None
+            # BNF12 is represented as a struct with 12 U384
 
-            coeffs = [
-                uint384_to_int(
-                    value[f"c{i}"]["d0"],
-                    value[f"c{i}"]["d1"],
-                    value[f"c{i}"]["d2"],
-                    value[f"c{i}"]["d3"],
-                )
-                for i in range(12)
-            ]
+            # In python, BNF12 is a tuple of 12 int but in cairo it's a struct with 12 U384
+            # Cast the U384 to int to be able to serialize
+            coeffs = [int(value[f"c{i}"]) for i in range(12)]
 
             return BNF12(tuple(coeffs))
 
         if python_cls == BNP12:
             # BNP12 is represented as a struct with two BNF12 values (x and y)
-            if value is None:
-                return None
-            # Create a new BNP12 point with these coordinates
             return BNP12(value["x"], value["y"])
```

Comment:
you should be able to remove this entirely, as this is supposed to be caught by the fallback case